gateway-ha/src/main/java/io/trino/gateway/ha/security/LbAuthenticator.java

```
@@ -51,7 +51,12 @@ public Optional<LbPrincipal> authenticate(String idToken)
         Optional<String> privilegesField = oauthManager.getPrivilegesField();
         if (privilegesField.isPresent()) {
             Map<String, Claim> claims = oauthManager.getClaimsFromIdToken(idToken).orElseThrow();
-            String userId = claims.get(userIdField).asString().replace("\"", "");
+            Claim userIdClaim = claims.get(userIdField);
+            if (userIdClaim == null) {
+                log.error("Required userId field %s not found", userIdField);
+                throw new AuthenticationException("UserId field does not exist");
```

The error log looks redundant if we throw an exception.

---

Thanks for reviewing. AFAIK, it does not appear twice as the exception gets suppressed in ChainedAuthFilter.

---

What is the expected action when users see this error?

---

@ebyhr they can correct the userIdField by decoding their token, similar to what privilegeField has. Logs for the privilegesField was extremely useful which motivated me to create this PR.
In my case, UI was working fine but it was failing for the API requests as the token used for the API request dont have email in it which was generated by the service.